packages/fiber/src/core/loop.ts

```diff
@@ -37,26 +37,22 @@ function run(effects: GlobalRenderCallback[], timestamp: number) {
   for (i = 0; i < effects.length; i++) effects[i](timestamp)
 }
 
-let subscribers: Subscription[]
-let subscription: Subscription
-function render(timestamp: number, state: RootState, frame?: THREE.XRFrame) {
+function update(timestamp: number, state: RootState, frame?: THREE.XRFrame) {
   // Run local effects
   let delta = state.clock.getDelta()
   // In frameloop='never' mode, clock times are updated using the provided timestamp
   if (state.frameloop === 'never' && typeof timestamp === 'number') {
     delta = timestamp - state.clock.elapsedTime
     state.clock.oldTime = state.clock.elapsedTime
     state.clock.elapsedTime = timestamp
+  } else {
+    delta = Math.min(delta, state.internal.maxDelta)
```

Comment:
This might already be taken care of by something deeper in the stack, but it might be worth clamping the time here (instead of just using `Math.min`), because there may be situations where the delta ends up being negative (rewinding the system time is a popular "game hack"). If `clock.elapsedTime` already takes care of this, ignore this comment.

Reply:
I don't think I understand. Could you say more?

Reply:
I don't know how `elapsedTime` behaves when the user manually sets their system clock to an earlier time. If the value "jumps back" accordingly, this might lead to negative delta times, while in turn will lead to many very very weird effects in a game.

This, of course, is an extreme edge case, but tampering with the system time _is_ a popular way to try and mess with games, which is why delta times are typically clamped between `0` and a max value.